### DISCUSSION TITLE
Should BranchDayOfWeekOperator's type hints match WeekDay.validate_week_day?

### DISCUSSION THREAD

**Discussion by tuzonghua** [apache/airflow#22247] [General]
In the BranchDayOfWeekOperator, the type hint for `week_day` is `Union[str, Iterable[str]]` ([source](https://github.com/apache/airflow/blob/main/airflow/operators/weekday.py#L54)), but in `WeekDay.validate_week_day` it's `Union[str, "WeekDay", Iterable[str], Iterable["WeekDay"]]` ([source](https://github.com/apache/airflow/blob/main/airflow/utils/weekday.py#L59)). I'm not sure if this is intentional or a bug, but if it's the latter I'm happy to open a small PR to fix.

**Comment by josh-fell** [comment]
Feel free to submit a PR to change it!
